### PR TITLE
Tests | Performance Tests Initial base

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Configure `runnerconfig.json` file with connection string and preferred settings to run Benchmark Jobs.
Run all enabled tests with below command:

```
cd src\Microsoft.Data.SqlClient\tests\PerformanceTests
dotnet run -c Release -f netcoreapp3.1|net5.0
```
_Only "**Release** Configuration" applies to Performance Tests_

### CODE REVIEW

**Comment by Wraith2** [comment]
How are these going to work given that we're not allowed to publish or compare any sql server benchmark numbers due to licensing?

**Comment by cheenamalhotra** [comment]
As of this test project, it's intended to be for development/review ease, we're not planning to publish benchmarks yet. We'll definitely go for legal approvals before doing that. Perf Project would simply live here for benchmarking aid.

/cc @David-Engel

**Comment by Wraith2** [comment]
Don't expect any answer from legal. They've been "working on it" for almost 3 years now with respect to TE benchmarking
https://github.com/TechEmpower/FrameworkBenchmarks/issues/3786

**Comment by cheenamalhotra** [comment]
FYI as per our queries with legal team, Microsoft products and teams are pre-approved for performance benchmarking and publishing. Community and customers however would need to seek approval to do so, and licensing restrictions apply to anyone non-Microsoft.

We will update docs accordingly and can proceed with this PR.

**Comment by Wraith2** [comment]
Is there a way for community members to get approved?

**Comment by cheenamalhotra** [comment]
That is a tough area, SQL Server owner team can only provide official approval. It seems Microsoft has been involved in TechEmpower/FrameworkBenchmarks#3786 to seek approval, but is still in progress. They are the first links IMO.

**Comment by Wraith2** [comment]
If you scroll to the bottom of the thread you linked you'll find the last post is by me, and it commemorates the third anniversary of the thread being opened and there is still no official response. Which is why I asked, I'm wondering if there's another way to contact someone because that thread has disappeared into internal bureaucracy and has become a delocalized question that no-one owns and everyone is waiting for someone else on.

If the perf of this library isn't adequate to put onto the TE charts then that can be worked on. As I know you're well aware we've made some excellent performance enhancements and we know there's more we can do with time and hard work.
If there are further improvements needed then we can work on them if we know what they are, refusing to participate at all just makes it look as if the results are known to be bad and means people just won't be interested in a product that seems not to have the confidence of it's owner behind it.

**Comment by cheenamalhotra** [comment]
We shall try to change this perception to best of our capacity @Wraith2 .
I can't speak for the past, but our team's present goals are including perf benchmarking and we do value driver performance and would like to target it hard for better user experience.

It will take a bit of work and time to come up with a proper process, but we're beginning it soon!
So let's give it a fresh start :)

**Review by DavoudEshtehari — APPROVED** [review]
Nice job 🎉

**Review by JRahnama — APPROVED** [review]

File: src/Microsoft.Data.SqlClient.sln
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient.sln:1228]: These are required as we add this project to solution. Will be cleaned up a bit with #1161 after merge.

File: src/Microsoft.Data.SqlClient/tests/PerformanceTests/Config/DataTypes.cs
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/PerformanceTests/Config/DataTypes.cs:35]: We want to keep it _object_ as actual types are not known from derivatives of "DataType", e.g. **MinMaxType** applies to all "Numerics" from Bit to Long. So keeping it long is for every type in between is not ideal.

File: src/Microsoft.Data.SqlClient/tests/PerformanceTests/DBFramework/Column.cs
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/PerformanceTests/DBFramework/Column.cs:10]: As Type property is not known here, the datatype of "Value" is also linked to it.

**Review by johnnypham — APPROVED** [review]

**Pull request merged**
